src/hooks/useComponents.tsx

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  SecurityComponents,
+  ErrorComponent,
+  LoadingElement,
+  IOktaContext,
+} from '../types';
+import OktaError from '../components/OktaError';
+
+export type ComponentsOptions = SecurityComponents;
+
+export interface ComponentsHook {
+  ErrorReporter: ErrorComponent;
+  Loading: LoadingElement | null;
+}
+
+const useComponents = (
```

Comment:
I don't think this needs to be written as a hook, it's really just a util function